bindings/cs/rl.net/LiveModel.cs

```diff
@@ -72,48 +72,108 @@ public bool TryInit(ApiStatus apiStatus = null)
             return result == NativeMethods.SuccessStatus;
         }
 
-        public bool TryChooseRank(string actionId, string contextJson, ActionFlags flags, out RankingResponse response, ApiStatus apiStatus = null)
+        public RankingResponse ChooseRank(string eventId, string contextJson)
+        {
+            ApiStatus apiStatus = new ApiStatus();
+            RankingResponse result = new RankingResponse();
+            if (!this.TryChooseRank(eventId, contextJson, result, apiStatus))
+            {
+                throw new RLException(apiStatus);
+            }
+
+            return result;
+        }
+
+        public bool TryChooseRank(string eventId, string contextJson, ActionFlags flags, out RankingResponse response, ApiStatus apiStatus = null)
         {
             response = new RankingResponse();
-            return this.TryChooseRank(actionId, contextJson, flags, response, apiStatus);
+            return this.TryChooseRank(eventId, contextJson, flags, response, apiStatus);
         }
 
-        public bool TryChooseRank(string actionId, string contextJson, ActionFlags flags, RankingResponse response, ApiStatus apiStatus = null)
+        public bool TryChooseRank(string eventId, string contextJson, ActionFlags flags, RankingResponse response, ApiStatus apiStatus = null)
         {
-            int result = LiveModelChooseRankWithFlags(this.NativeHandle, actionId, contextJson, (uint)flags, response.NativeHandle, apiStatus.ToNativeHandleOrNullptr());
+            int result = LiveModelChooseRankWithFlags(this.NativeHandle, eventId, contextJson, (uint)flags, response.NativeHandle, apiStatus.ToNativeHandleOrNullptr());
             return result == NativeMethods.SuccessStatus;
         }
 
-        public bool TryReportActionTaken(string actionId, ApiStatus apiStatus = null)
+        public RankingResponse ChooseRank(string eventId, string contextJson, ActionFlags flags)
         {
-            int result = LiveModelReportActionTaken(this.NativeHandle, actionId, apiStatus.ToNativeHandleOrNullptr());
+            ApiStatus apiStatus = new ApiStatus();
+            RankingResponse result = new RankingResponse();
+            if (!this.TryChooseRank(eventId, contextJson, flags, result, apiStatus))
+            {
+                throw new RLException(apiStatus);
+            }
+
+            return result;
+        }
+
+        public bool TryReportActionTaken(string eventId, ApiStatus apiStatus = null)
+        {
+            int result = LiveModelReportActionTaken(this.NativeHandle, eventId, apiStatus.ToNativeHandleOrNullptr());
             return result == NativeMethods.SuccessStatus;
         }
 
-        public bool TryReportOutcome(string actionId, float outcome, ApiStatus apiStatus = null)
+        public void ReportActionTaken(string eventId)
+        {
+            ApiStatus apiStatus = new ApiStatus();
```

Comment:
We should be disposing the temporary ApiStatus objects.

Reply:
Good call.